File: Android/sources/com/android/billingclient/api/ProxyBillingActivityV2.java
package com.android.billingclient.api;

import android.app.PendingIntent;
import android.content.Intent;
import android.os.Bundle;
import android.os.ResultReceiver;
import com.google.android.apps.common.proguard.UsedByReflection;

@UsedByReflection("PlatformActivityProxy")
public class ProxyBillingActivityV2 extends bc3 {
    private y8 zza;
    private y8 zzb;
    private ResultReceiver zzc;
    private ResultReceiver zzd;

    public final void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        this.zza = registerForActivityResult(new x8(2), new zzcv(this));
        this.zzb = registerForActivityResult(new x8(2), new zzcw(this));
        if (bundle == null) {
            oyg.e("ProxyBillingActivityV2", "Launching Play Store billing dialog");
            if (getIntent().hasExtra("ALTERNATIVE_BILLING_ONLY_DIALOG_INTENT")) {
                this.zzc = (ResultReceiver) getIntent().getParcelableExtra("alternative_billing_only_dialog_result_receiver");
                this.zza.a(new ui7(((PendingIntent) getIntent().getParcelableExtra("ALTERNATIVE_BILLING_ONLY_DIALOG_INTENT")).getIntentSender(), (Intent) null, 0, 0));
            } else if (getIntent().hasExtra("external_payment_dialog_pending_intent")) {
                this.zzd = (ResultReceiver) getIntent().getParcelableExtra("external_payment_dialog_result_receiver");
                this.zzb.a(new ui7(((PendingIntent) getIntent().getParcelableExtra("external_payment_dialog_pending_intent")).getIntentSender(), (Intent) null, 0, 0));
            }
        } else if (bundle.containsKey("alternative_billing_only_dialog_result_receiver")) {
            this.zzc = (ResultReceiver) bundle.getParcelable("alternative_billing_only_dialog_result_receiver");
        } else if (bundle.containsKey("external_payment_dialog_result_receiver")) {
            this.zzd = (ResultReceiver) bundle.getParcelable("external_payment_dialog_result_receiver");
        }
    }

    public final void onSaveInstanceState(Bundle bundle) {
        super.onSaveInstanceState(bundle);
        ResultReceiver resultReceiver = this.zzc;
        if (resultReceiver != null) {
            bundle.putParcelable("alternative_billing_only_dialog_result_receiver", resultReceiver);
        }
        ResultReceiver resultReceiver2 = this.zzd;
        if (resultReceiver2 != null) {
            bundle.putParcelable("external_payment_dialog_result_receiver", resultReceiver2);
        }
    }

    public final void zza(u8 u8Var) {
        Intent intent = u8Var.b;
        int responseCode = oyg.c(intent, "ProxyBillingActivityV2").getResponseCode();
        ResultReceiver resultReceiver = this.zzc;
        if (resultReceiver != null) {
            resultReceiver.send(responseCode, intent == null ? null : intent.getExtras());
        }
        int i = u8Var.a;
        finish();
    }

    public final void zzb(u8 u8Var) {
        Intent intent = u8Var.b;
        int responseCode = oyg.c(intent, "ProxyBillingActivityV2").getResponseCode();
        ResultReceiver resultReceiver = this.zzd;
        if (resultReceiver != null) {
            resultReceiver.send(responseCode, intent == null ? null : intent.getExtras());
        }
        int i = u8Var.a;
        if (!(i == -1 && responseCode == 0)) {
            StringBuilder sb = new StringBuilder("External offer dialog finished with resultCode: ");
            sb.append(i);
            sb.append(" and billing's responseCode: ");
            sb.append(responseCode);
        }
        finish();
    }
}

File: IOS/v2/com/android/billingclient/api/ProxyBillingActivityV2.swift
package com.iOS.billingclient.api;.swift
import iOS.app.PendingIntent;.swift
import iOS.content.Intent;.swift
import iOS.os.Bundle;.swift
import iOS.os.ResultReceiver;.swift
import com.google.iOS.apps.common.proguard.UsedByReflection;.swift
@UsedByReflection("PlatformActivityProxy").swift
public class ProxyBillingActivityV2 extends bc3 {.swift
private y8 zza;.swift
private y8 zzb;.swift
private ResultReceiver zzc;.swift
private ResultReceiver zzd;.swift
public final void onCreate(Bundle bundle) {.swift
super.onCreate(bundle);.swift
this.zza = registerForActivityResult(new x8(2), new zzcv(this));.swift
this.zzb = registerForActivityResult(new x8(2), new zzcw(this));.swift
if (bundle == null) {.swift
oyg.e("ProxyBillingActivityV2", "Launching Play Store billing dialog");.swift
if (getIntent().hasExtra("ALTERNATIVE_BILLING_ONLY_DIALOG_INTENT")) {.swift
this.zzc = (ResultReceiver) getIntent().getParcelableExtra("alternative_billing_only_dialog_result_receiver");.swift
this.zza.a(new ui7(((PendingIntent) getIntent().getParcelableExtra("ALTERNATIVE_BILLING_ONLY_DIALOG_INTENT")).getIntentSender(), (Intent) null, 0, 0));.swift
} else if (getIntent().hasExtra("external_payment_dialog_pending_intent")) {.swift
this.zzd = (ResultReceiver) getIntent().getParcelableExtra("external_payment_dialog_result_receiver");.swift
this.zzb.a(new ui7(((PendingIntent) getIntent().getParcelableExtra("external_payment_dialog_pending_intent")).getIntentSender(), (Intent) null, 0, 0));.swift
}.swift
} else if (bundle.containsKey("alternative_billing_only_dialog_result_receiver")) {.swift
this.zzc = (ResultReceiver) bundle.getParcelable("alternative_billing_only_dialog_result_receiver");.swift
} else if (bundle.containsKey("external_payment_dialog_result_receiver")) {.swift
this.zzd = (ResultReceiver) bundle.getParcelable("external_payment_dialog_result_receiver");.swift
}.swift
}.swift
public final void onSaveInstanceState(Bundle bundle) {.swift
super.onSaveInstanceState(bundle);.swift
ResultReceiver resultReceiver = this.zzc;.swift
if (resultReceiver = null) {.swift
bundle.putParcelable("alternative_billing_only_dialog_result_receiver", resultReceiver);.swift
}.swift
ResultReceiver resultReceiver2 = this.zzd;.swift
if (resultReceiver2 = null) {.swift
bundle.putParcelable("external_payment_dialog_result_receiver", resultReceiver2);.swift
}.swift
}.swift
public final void zza(u8 u8Var) {.swift
Intent intent = u8Var.b;.swift
int responseCode = oyg.c(intent, "ProxyBillingActivityV2").getResponseCode();.swift
ResultReceiver resultReceiver = this.zzc;.swift
if (resultReceiver = null) {.swift
resultReceiver.send(responseCode, intent == null ? null : intent.getExtras());.swift
}.swift
int i = u8Var.a;.swift
finish();.swift
}.swift
public final void zzb(u8 u8Var) {.swift
Intent intent = u8Var.b;.swift
int responseCode = oyg.c(intent, "ProxyBillingActivityV2").getResponseCode();.swift
ResultReceiver resultReceiver = this.zzd;.swift
if (resultReceiver = null) {.swift
resultReceiver.send(responseCode, intent == null ? null : intent.getExtras());.swift
}.swift
int i = u8Var.a;.swift
if ((i == -1 && responseCode == 0)) {.swift
StringBuilder sb = new StringBuilder("External offer dialog finished with resultCode: ");.swift
sb.append(i);.swift
sb.append(" and billing's responseCode: ");.swift
sb.append(responseCode);.swift
}.swift
finish();.swift
}.swift
}.swift
